batch/batch/driver/instance_collection/pool.py

```diff
@@ -334,25 +350,42 @@ async def ready_cores_mcpu_from_estimated_job_queue(self):
     LIMIT {share * JOB_QUEUE_SCHEDULING_WINDOW_SECONDS}
   )
 ) AS t
+ORDER BY batch_id ASC, always_run DESC, job_id ASC
 LIMIT {share * JOB_QUEUE_SCHEDULING_WINDOW_SECONDS}
 )
 '''
 
             jobs_query.append(user_job_query)
             jobs_query_args += [user, self.name, user, self.name]
 
-        result = await self.db.select_and_fetchone(
+        result = self.db.select_and_fetchall(
+            # We use the string representation of regions JSON_UNQUOTE for order by instead of the json array
```

Comment:
I'm not sure how you would do that. 

I'm confused myself now about why the return type of the query is not just a Python list. The UNQUOTE is because you can't order by JSON lists in MySQL. Otherwise, it shouldn't be necessary. I will need to convert the regions list into a string for the prometheus metrics though.

Reply:
The return type of a JSON object is actually a string that our code then needs to parse into a Python object.

I approached the prometheus metrics by only including one region and using a weighted average of the statistic across the regions selected for each job.